### PR TITLE
feat: apply gcache to be used for authentication

### PR DESCRIPTION
**Pull request by gyuguen**
A few cache-related features have been added.
* Here, [gcache](https://github.com/bluele/gcache) is used to store the per-request authentication information
  * gcache provides a maximum cache size and data deletion algorithms.
* TTL and total size of the cache are separated by settings.

### CODE REVIEW

**Review by hansol-medi — APPROVED** [review]
LGTM

File: config/toml.go [hunk 1]
```diff
@@ -60,6 +60,15 @@ secret-access-key = "{{ .AWSS3.SecretAccessKey }}"
 enable = {{ .Enclave.Enable }}
 attestation-provider-addr = "{{ .Enclave.AttestationProviderAddr }}"
 
+###############################################################################
+###                         AuthenticateConfig AuthenticateConfig Configuration                           ###
```
Comment (youngjoon-lee) [inline config/toml.go:64]: pls organize this comment :)

File: config/toml.go [hunk 2]
```diff
@@ -60,6 +60,15 @@ secret-access-key = "{{ .AWSS3.SecretAccessKey }}"
...
+###############################################################################
+
+[authenticate]
+
+expiration = "{{ .Authenticate.Expiration }}"
+size = {{ .Authenticate.Size }}
```
Comment (youngjoon-lee) [inline config/toml.go:70]: pls rename authenticate to authentication or auth, at other source codes as well.

**Review by youngjoon-lee — APPROVED** [review]
lgtm

**Pull request merged**